release.sh

```diff
@@ -28,15 +28,17 @@ check() {
         exit 1
     fi
     while read line; do
-        sha=$(echo "$line" | gawk '{print $1}')
-        ver=$(echo "$line" | gawk '{print $2}')
+        local sha=$(echo "$line" | gawk '{print $1}')
```

Comment:
Prefixing a variable with `local` (or `export`) suppresses any errors during RHS evaluation.

```bash
$ function foo { VAR=$(false); echo $?; }; foo
1
$ function foo { local VAR=$(false); echo $?; }; foo
0
```

If you need to crash if this line fails, then you might need to separate the `local` part. I tend to just move it to the top of the file, C-style:

```
check() {
    local sha ver ver_sha
    # rest of function
}
```

Reply:
Thanks for the tip, I did not know that. I'll hoist them as suggested.

Reply:
I learnt it from `shellcheck`. 🙂